docs/modules/serialization/pages/schema-evolution.adoc

```diff
@@ -0,0 +1,703 @@
+
+| `Fingerprint`
+| A 64-bit identifier derived from the schema definition using a link:https://en.wikipedia.org/wiki/Rabin_fingerprint[Rabin Fingerprint] algorithm. Any change to a schema produces a new fingerprint.
+
+| `Backward Compatibility`
+| The ability of newer code (reader) to correctly read data serialized by older code (writer).
+
+| `Forward Compatibility`
+| The ability of older code (reader) to correctly read data serialized by newer code (writer), ignoring unknown fields.
+
+| `FieldKind`
+| The internal representation of a field’s data type in Compact serialization (e.g., `INT32`, `STRING`, `DECIMAL`, `ARRAY_OF_INT64`).
+
+|===
+
+[NOTE]
+====
+The following information applies to cases where Compact serialization is used. There's no guarantee that it works when Java Serialization or other supported serialization mechanisms are used
+====
+
+== Core principles and strategies
+
+Schema evolution in Hazelcast is guided by a set of core principles designed to ensure that data changes remain predictable, compatible, and easy to manage across rolling upgrades, mixed client versions, and persistent clusters.
+
+=== Stable `typeName`
+
+Each Compact type is identified by a `typeName`, which must remain stable throughout the lifetime of a class.
+The `typeName` forms the logical link between different schema versions of the same object. Changing it effectively creates a new Compact type, isolating the data written under the old schema.
+
+Maintaining a stable `typeName` allows new and old members to exchange data transparently, because Hazelcast treats each version of the same `typeName` as an evolution rather than a different entity.
+
+=== Deterministic writes
+
+Hazelcast derives a schema fingerprint from the fields written in the serializer’s `write` method.
+The structure and order of these writes must be **deterministic** — meaning they do not depend on runtime conditions or object state.
+
+Conditional writes, skipped fields, or dynamic field names can lead to multiple fingerprints being generated for logically identical objects, causing unexpected deserialization errors or redundant schema registrations.
+
+To ensure deterministic writes:
+
+* Always write all defined fields, even if their values are `null` or default.
+* Avoid conditional or state-dependent logic inside the `write` method.
+* Keep field names and their corresponding `FieldKind` stable between releases.
+
+=== Schema evolution approaches
+
+Hazelcast supports two primary approaches to managing schema evolution, depending on operational needs and compatibility requirements.
+
+[cols="1,3", options="header"]
+|===
+| Approach | Description
+
+| **Single Map Evolution**
+| All schema changes occur within the same map and `typeName`. This approach allows members to coexist using different schema versions during rolling upgrades. It is suitable for additive or compatible changes (for example, adding new fields, widening types).
```

Comment:
```suggestion
| All schema changes occur within the same IMap and `typeName`. This approach allows members to coexist using different schema versions during rolling upgrades. It is suitable for additive or compatible changes (for example, adding new fields).
```
Widening types or removing a field is forward incompatible if the old reader does not do field existence check (which is usually not the case).